agent/src/main/java/com/microsoft/hydralab/agent/service/AgentWebSocketClientService.java

```diff
@@ -160,13 +162,19 @@ public void onMessage(Message message) {
 
     private void heartbeatResponse(Message message) {
         AgentMetadata agentMetadata = (AgentMetadata) message.getBody();
-        blobStorageClient.setSASData(agentMetadata.getBlobSAS());
+
+        storageTypeConsistencyCheck(agentMetadata.getAccessToken());
+        storageServiceClient.updateAccessToken(agentMetadata.getAccessToken());
         syncAgentStatus(agentMetadata.getAgentUser());
-        if (isPrometheusEnabled && !pushGateway.isBasicAuthSet.get()) {
-            pushGateway.setConnectionFactory(new BasicAuthHttpConnectionFactory(agentMetadata.getPushgatewayUsername(), agentMetadata.getPushgatewayPassword()));
-            ThreadUtils.safeSleep(1000);
-            pushGateway.isBasicAuthSet.set(true);
-            log.info("Pushgateway has set basic auth now, data can be pushed correctly.");
+        prometheusPushgatewayInit(agentMetadata);
+    }
+
+    private void storageTypeConsistencyCheck(AccessToken accessToken) {
+        try {
+            storageServiceClient.storageTypeCheck(accessToken);
+        } catch (HydraLabRuntimeException e) {
+            log.error(e.getMessage());
+            System.exit(e.getCode());
```

Comment:
Why exit the system here?

Reply:
Added StorageManageService for StorageServiceClient init, remove storage type check related code.